app/src/main/java/nerd/tuxmobil/fahrplan/congress/base/SessionsAdapter.kt

```diff
@@ -0,0 +1,213 @@
+
+        if (convertView == null) {
+            // clone the inflater using the ContextThemeWrapper
+            val inflater = context.getLayoutInflater()
+            val localInflater = inflater.cloneInContext(context)
+
+            when (type) {
+                TYPE_ITEM -> {
+                    rowView = localInflater.inflate(R.layout.session_list_item, parent, false)
+                    viewHolder = ViewHolder(
+                        rowView.requireViewByIdCompat(R.id.session_list_item_title_view),
+                        rowView.requireViewByIdCompat(R.id.session_list_item_subtitle_view),
+                        rowView.requireViewByIdCompat(R.id.session_list_item_speakers_view),
+                        rowView.requireViewByIdCompat(R.id.session_list_item_language_view),
+                        rowView.requireViewByIdCompat(R.id.session_list_item_day_view),
+                        rowView.requireViewByIdCompat(R.id.session_list_item_time_view),
+                        rowView.requireViewByIdCompat(R.id.session_list_item_room_view),
+                        rowView.requireViewByIdCompat(R.id.session_list_item_duration_view),
+                        rowView.requireViewByIdCompat(R.id.session_list_item_video_view),
+                        rowView.requireViewByIdCompat(R.id.session_list_item_no_video_view),
+                        rowView.requireViewByIdCompat(R.id.session_list_item_without_video_recording_view),
+                    )
+                    rowView.tag = viewHolder
+                }
+                TYPE_SEPARATOR -> {
+                    rowView = localInflater.inflate(R.layout.session_list_separator, parent, false)
+                    viewHolderSeparator = ViewHolderSeparator()
+                    viewHolderSeparator.text = rowView.requireViewByIdCompat(R.id.session_list_separator_title_view)
+                    rowView.tag = viewHolderSeparator
+                }
+                else -> {
+                    error("Unknown type: $type")
+                }
+            }
+        } else {
+            rowView = convertView
+            when (type) {
+                TYPE_ITEM -> {
+                    viewHolder = rowView.tag as ViewHolder
+                }
+                TYPE_SEPARATOR -> {
+                    viewHolderSeparator = rowView.tag as ViewHolderSeparator
+                }
+                else -> {
+                    error("Unknown type: $type")
+                }
+            }
+        }
+        when (type) {
+            TYPE_ITEM -> {
+                setItemContent(position, viewHolder!!)
+            }
+            TYPE_SEPARATOR -> {
+                setSeparatorContent(position, viewHolderSeparator!!)
+            }
+            else -> {
+                error("Unknown type: $type")
+            }
+        }
+        return rowView
+    }
+
+    fun getSession(position: Int) = list[mapper[position]]
+
+    protected open fun resetItemStyles(viewHolder: ViewHolder) {
```

Comment:
Why `open`?

Reply:
`open` means that other classes can inherit this function. Normally you would make it `abstract`, but that is only possible if the function does not have a body.

Reply:
This function is called from the child classes but they do not overwrite the functionality. The `open` is not needed here. I removed it.